scripts/d.correlate/d.correlate.py

```diff
@@ -98,12 +98,12 @@ def main():
                 p = gcore.feed_command('d.graph', color=color)
                 ofile = p.stdin
 
-                ifile = file(tmpfile, 'r')
+                ifile = open(tmpfile, 'r')
                 for l in ifile:
                     f = l.rstrip('\r\n').split(' ')
                     x = float(f[0])
                     y = float(f[1])
-                    ofile.write("icon + 0.1 %f %f\n" % ((x - minx + 1) * kx,
+                    ofile.write(b"icon + 0.1 %f %f\n" % ((x - minx + 1) * kx,
```

Comment:
I haven't actually test your changes, but this seems like it won't work in Python 3, you opened the file in text mode and you write there bytestring, so you should be getting:
`TypeError: write() argument must be str, not bytes`.

Reply:
Ciao Anna. I just changed `file()` to `open()`. Then,
```
python --version
Python 3.8.5
```
and
```
→ d.correlate map=S1_GRD_377344.171763875_5556366.58391489_602106.99615688_5327454.6501018_ASCENDING_20190610_20190617_median_VV,S1_GRD_377344.171763875_5556366.58391489_602106.99615688_5327454.6501018_DESCENDING_20190610_20190617_median_VV
 100%
Traceback (most recent call last):
  File "/osgeo/grass/dist.x86_64-pc-linux-gnu/scripts/d.correlate", line 119, in <module>
    sys.exit(main())
  File "/osgeo/grass/dist.x86_64-pc-linux-gnu/scripts/d.correlate", line 106, in main
    ofile.write("icon + 0.1 %f %f\n" % ((x - minx + 1) * kx,
TypeError: a bytes-like object is required, not 'str'
```
then, changing from
```
                    ofile.write("icon + 0.1 %f %f\n" % ((x - minx + 1) * kx,
```
to
```
                    ofile.write(b"icon + 0.1 %f %f\n" % ((x - minx + 1) * kx,
```
works here
```
→ d.correlate map=S1_GRD_377344.171763875_5556366.58391489_602106.99615688_5327454.6501018_ASCENDING_20190610_20190617_median_VV,S1_GRD_377344.171763875_5556366.58391489_602106.99615688_5327454.6501018_DESCENDING_20190610_20190617_median_VV
 100%
```
and I get an image output too.

Reply:
Sorry, I overlooked `ofile` is redefined on line 99, that makes sense now.